### PR TITLE
Realize column factory

### PR DESCRIPTION
**Pull request by Tigrov**
### Related PRs
- yiisoft/db-sqlite#314
- yiisoft/db-mysql#346
- yiisoft/db-pgsql#355
- yiisoft/db-oracle#276
- yiisoft/db-mssql#316

| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | #854

### CODE REVIEW

File: src/Schema/Column/ColumnFactory.php [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+use const PHP_INT_SIZE;
+
+/**
+ * The default implementation of the {@see ColumnFactoryInterface}.
+ *
+ * @psalm-import-type ColumnInfo from ColumnSchemaInterface
+ * @psalm-suppress MixedArgumentTypeCoercion
+ */
+class ColumnFactory implements ColumnFactoryInterface
```
Comment (vjik) [inline src/Schema/Column/ColumnFactory.php:26]:
```suggestion
abstract class AbstractColumnFactory implements ColumnFactoryInterface
```

All DBMS inherit column factory. Suggest mark this class abstract.

File: src/Schema/Column/ColumnFactoryInterface.php [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+namespace Yiisoft\Db\Schema\Column;
+
+/**
+ * The interface must be implemented by a column factory class. It should create a column schema for a database column
+ * type and initialize column information.
+ *
+ * @psalm-import-type ColumnInfo from ColumnSchemaInterface
+ */
+interface ColumnFactoryInterface
```
Comment (vjik) [inline src/Schema/Column/ColumnFactoryInterface.php:13]:
```suggestion
interface ColumnSchemaFactoryInterface
```

Comment (Tigrov) [reply src/Schema/Column/ColumnFactoryInterface.php:13]: The word `Schema` is planned to be removed from the names of classes associated with columns

File: src/Schema/Column/ColumnFactory.php (hunk 1 above)
Comment (Tigrov) [reply src/Schema/Column/ColumnFactory.php:26]: `ColumnBuilder` will be made to generate columns when creating tables, for example in `db-migration` package or in `db` package. Regardless of the specific DBMS. For this, an independent implementation of `ColumnFactory` is needed.

Comment (Tigrov) [reply src/Schema/Column/ColumnFactory.php]: Perhaps make sense to realize separate implementation from `AbstractColumnFactory` for this purpose.

**Review by vjik — APPROVED** [review]

**Pull request merged**